dvc/scm.py

```diff
@@ -123,3 +125,56 @@ def resolve_rev(scm: "Git", rev: str) -> str:
             if len(ref_infos) > 1:
                 raise RevError(f"ambiguous Git revision '{rev}'")
         raise RevError(str(exc))
+
+
+def iter_revs(
+    scm: "Git",
+    head_revs: Optional[List[str]] = None,
+    num: int = 1,
+    all_branches: bool = False,
+    all_tags: bool = False,
+    all_commits: bool = False,
+    all_experiments: bool = False,
+) -> Mapping[str, List[str]]:
+    from dvc.repo.experiments.utils import fix_exp_head
+
+    if num < 1 and num != -1:
+        raise InvalidArgumentError(f"Invalid number of commits '{num}'")
+
+    if not any(
+        [head_revs, all_branches, all_tags, all_commits, all_experiments]
+    ):
+        return {}
+
+    head_revs = head_revs or []
+    revs = []
+    for rev in head_revs:
```

Comment:
Wasn't this mutually exclusive with `num`/`n` before?

Reply:
Can't this be done separately with `n_commits(num)` if it's exclusive, outside of `iter_revs`?

Reply:
In the previous version `rev+num` is not mutually exclusive with `all_tags/all_branches`. And the result will be the sum of all revs. Maybe it is better to make them exclusive? @dberenbaum 
This function will be used in `exp push/pull/ls/show/remove` If I separate `n_commits(num)` here then I need to add a new function and call them five times there.

Reply:
> And the result will be the sum of all revs.

How is this supposed to work? If I do `dvc exp show -a -n 10`, I get:

```
 ───────────────────────────────────────────────────────────────────────────────────────
  Experiment                     Created        avg_prec   roc_auc   prepare.split   pre
 ───────────────────────────────────────────────────────────────────────────────────────
  workspace                      -               0.60405    0.9608   0.2             201
  11-random-forest-experiments   May 29, 2021    0.60405    0.9608   0.2             201
 ───────────────────────────────────────────────────────────────────────────────────────
```

So it seems like `-n` is being ignored?

I don't feel strongly about whether we allow these to be combined. It seems strange to combine them, but if we can explain the behavior clearly, it might be fine to allow users to combine them if they want.